library/proc_macro/src/bridge/client.rs

```diff
@@ -37,10 +19,22 @@ macro_rules! define_handles {
         }
 
         impl<S: server::Types> HandleStore<S> {
-            pub(super) fn new(handle_counters: &'static HandleCounters) -> Self {
+            pub(super) fn new() -> Self {
+                // FIXME(eddyb) these counters are server-side, so they don't
+                // protect against the same proc macro dylib being used with
+                // multiple servers - however, that's very unlikely, and should
+                // be protected against through other means (e.g. forcing a
+                // specific proc macro dylib to always talk to the same server
+                // that initially loaded it).
```

Comment:
Perhaps mention "the same dylib" or something like that? This could be interpreted as "the `proc_macro_server::Rustc` object should be the same for every invocation of a macro from each crate", which I don't think is what you're suggesting (as ideally I'd imagine that we we want to throw out `HandleStore` after each invocation of a proc macro, and only keep around the handle counter to avoid conflicts, which is done here by using a static for the counter).